openff/toolkit/topology/topology.py

```diff
@@ -2308,51 +1464,68 @@ def to_openmm(self, ensure_unique_atom_names=True):
         # We need to iterate over the topology molecules if we want to
         # keep track of chains/residues as Atom.topology_molecule is
         # instantiated every time and can't be used as a key.
-        for topology_molecule in self.topology_molecules:
-            for atom in topology_molecule.atoms:
-                reference_molecule = topology_molecule.reference_molecule
-                n_molecules = len(
-                    self._reference_molecule_to_topology_molecules[reference_molecule]
-                )
-
-                # Add 1 chain per molecule unless there are more than 5 copies,
-                # in which case we add a single chain for all of them.
-                if n_molecules <= 5:
-                    # We associate a chain to each molecule.
-                    key_molecule = topology_molecule
-                else:
-                    # We associate a chain to all the topology molecule.
-                    key_molecule = reference_molecule
+        for molecule in self.molecules:
+            molecule_id = self.molecule_index(molecule)
+            residue_id = molecule.name
+            for atom in molecule.atoms:
+                # reference_molecule = topology_molecule#.reference_molecule
```

Comment:
I'm working on refactoring this logic to "play nice" with our in-built hierarchy so I'd like to keep the "old way" commented out here so I can refer back to it.